ui/app/AppLayouts/Profile/helpers/ContactModelEntry.qml

```diff
@@ -13,6 +13,15 @@ QObject {
     required property string publicKey
     required property var contactsModel
 
+    signal populateContactDetailsRequested()
+
+    onPublicKeyChanged: {
+        if (root.publicKey && contactsModel && !contactsModel.hasUser(root.publicKey)) {
```

Comment:
If we have a valid pubkey and contactsModel, we check if the model has that user. If it does not, we request that it gets populated.

This happens when we want the data from a user we never were able to fetch the data from Waku. 

So we either get it from the contacts cache or it will be generated (alias, emoji, color) from the pubkey.